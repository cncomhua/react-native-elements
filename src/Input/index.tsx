import { withTheme } from '../config';
import { Input, InputProps } from './Input';

export { Input };
export type { InputProps };
/**
 * RNE Input
 *
 * **Preview**
 *
 * ![URL](URL)
 *
 * [**Documentation**](https://reactnativeelements.com/docs/input)
 */
export default withTheme(Input, 'Input');
