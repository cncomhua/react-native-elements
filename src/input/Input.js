import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Text, View, TextInput, Dimensions } from 'react-native';
import ViewPropTypes from '../config/ViewPropTypes';
const SCREEN_WIDTH = Dimensions.get('window').width;

class Input extends Component {
  focus() {
    this.input.focus();
  }

  blur() {
    this.input.blur();
  }

  clear() {
    this.input.clear();
  }

  render() {
    const {
      containerStyle,
      icon,
      iconContainerStyle,
      inputStyle,
      displayError,
      errorStyle,
      errorMessage,
      ...attributes
    } = this.props;

    const styles = this.context.theme.input;

    return (
      <View>
        <View
          style={[
            styles.container,
            { width: SCREEN_WIDTH - 100, height: 40 },
            containerStyle,
          ]}
        >
          {icon &&
            <View
              style={[styles.iconContainer, { height: 40 }, iconContainerStyle]}
            >
              {icon}
            </View>}
          <TextInput
            ref={input => (this.input = input)}
            underlineColorAndroid="transparent"
            style={[
              styles.input,
              { width: SCREEN_WIDTH - 100, height: 40 },
              inputStyle,
            ]}
            {...attributes}
          />
        </View>
        {displayError &&
          <Text style={[styles.error, errorStyle && errorStyle]}>
            {errorMessage || 'Error!'}
          </Text>}
      </View>
    );
  }
}

Input.propTypes = {
  containerStyle: ViewPropTypes.style,
  icon: PropTypes.object,
  iconContainerStyle: ViewPropTypes.style,
  inputStyle: PropTypes.object,
  displayError: PropTypes.bool,
  errorStyle: PropTypes.object,
  errorMessage: PropTypes.string,
};

Input.contextTypes = {
  theme: PropTypes.object.isRequired,
};

export default Input;
