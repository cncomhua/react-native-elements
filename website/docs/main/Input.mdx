---
id: input
title: Input
slug: /input
---

import Usage from './usage/Input/Input.mdx';

import Props from './props/Input.mdx';

Inputs allow users to enter text into a UI. They typically appear in forms and
dialogs.

<div className="component-preview component-preview--grid">
  <figure>
    <img src="/img/input/input--placeholder.png" alt="Input with placeholder" />
    <figcaption>Placeholder</figcaption>
  </figure>
  <figure>
    <img src="/img/input/input--label.png" alt="Input with Label and icons" />
    <figcaption>Label with Icons</figcaption>
  </figure>
</div>

## Usage

<Usage />

---

## Props

<Props />
