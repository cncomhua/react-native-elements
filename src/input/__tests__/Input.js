import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from '../Input';

describe('Input Component', () => {
  it('should render without issues', () => {
    const component = shallow(<Input />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should set the name prop and give it back in onChangeText', () => {
    const onChangeTextMock = jest.fn();
    const component = shallow(
      <Input name="myInputName" onChangeText={onChangeTextMock} />
    );

    const event = expect.anything(); // TODO Do we have somewhere a defined structure for event?i
    component.find('TextInput').simulate('change', event);
    expect(onChangeTextMock).toHaveBeenCalledTimes(1);
    expect(onChangeTextMock).toBeCalledWith(event, { name: 'myInputName' });
  });
});
