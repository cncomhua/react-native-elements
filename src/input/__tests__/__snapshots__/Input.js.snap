// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render without issues 1`] = `
<Component
  style={
    Array [
      Object {
        "width": "90%",
      },
      undefined,
    ]
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
        },
        undefined,
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChange={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "height": 40,
            "marginLeft": 10,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </AnimatedComponent>
</Component>
`;
